department-of-veterans-affairs/next-build
# VACMS-10254 refactor benefit hubs

## Description
Closes #10254 https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10254

There's an extra commit or two due to rebasing weirdness.. not quite sure to be honest.

## Testing done
Brought back the demo page locally with this code, money line is this: 
```js
 const benefitsHubCollectionProps = benefitsHubCollection.map(
    (benefitsHub) => {
      return {
        title: benefitsHub.title,
        relatedBenefitHubs: benefitsHub.field_related_benefit_hubs.map((hub) =>
          queries.formatData('node--landing_page', hub)
        ),
      }
    }
  )
  ```
  
  full demo page code is this:
  ```js
import { drupalClient } from '@/lib/utils/drupalClient'
import { GetStaticPropsContext, GetStaticPropsResult } from 'next'
import Container from '@/templates/common/container'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import {
  NodeResourceType,
  NodeSupportResourcesDetailPage,
} from '@/types/dataTypes/drupal/node'
import { BenefitsHubLinks } from '@/templates/common/benefitsHubLinks'
import { queries } from '@/data/queries'

interface BenefitsHubLandingPageProps {
  benefitsHubCollectionProps: any
}

const BenefitsHubPage = ({
  benefitsHubCollectionProps,
}: BenefitsHubLandingPageProps) => {
  if (!benefitsHubCollectionProps) benefitsHubCollectionProps = []

  return benefitsHubCollectionProps.map((benefitsHubProp) => (
    <>
      <Container className="container">
        <BenefitsHubLinks {...benefitsHubProp} />
        <hr />
      </Container>
    </>
  ))
}
export default BenefitsHubPage

export async function getStaticProps(
  context: GetStaticPropsContext
): Promise<GetStaticPropsResult<BenefitsHubLandingPageProps>> {
  const params = new DrupalJsonApiParams()
  params
    .addFilter('status', '1')
    .addFilter('field_related_benefit_hubs', null, 'IS NOT NULL')
    .addSort('created', 'DESC')
    .addInclude(['field_related_benefit_hubs'])

  const benefitsHubCollection =
    await drupalClient.getResourceCollectionFromContext<
      NodeSupportResourcesDetailPage[]
    >(NodeResourceType.SupportResourcesDetailPage, context, {
      params: params.getQueryObject(),
    })

  const benefitsHubCollectionProps = benefitsHubCollection.map(
    (benefitsHub) => {
      return {
        title: benefitsHub.title,
        relatedBenefitHubs: benefitsHub.field_related_benefit_hubs.map((hub) =>
          queries.formatData('node--landing_page', hub)
        ),
      }
    }
  )

  return {
    props: {
      benefitsHubCollectionProps,
    },
  }
}

  ```

## Screenshots
As seen on http://localhost:3000/demo/benefitsHubLinks/
![Screen Shot 2022-09-28 at 12 27 44 PM](https://user-images.githubusercontent.com/11279744/192871478-036285eb-c7d2-4250-8b32-21cbb71b9162.png)


## QA steps
Add demo page code from above locally, visit http://localhost:3000/demo/benefitsHubLinks/ see that benefitHubs render as expected with the new query instead of the dataservice

### Definition of Done
